test/e2e/antreapolicy_test.go

```diff
@@ -4180,6 +4278,34 @@ func doProbe(t *testing.T, data *TestData, p *CustomProbe, protocol AntreaPolicy
 	}
 }
 
+func doEvaluation(t *testing.T, data *TestData, npEvaluation *NPEvaluation) {
+	if !testOptions.npEvaluation {
+		return
+	}
+	log.Infof("running networkpolicyevaluation tests")
+	antreaController, err := data.getAntreaController()
+	if err != nil {
+		t.Errorf("Error occurred when trying to get the Antrea Controller : %v", err)
+	}
+	antctlArgs := []string{"antctl", "query", "networkpolicyevaluation"}
+	getRealizedPod := func(label Pod) string {
+		realizedPod, err := k8sUtils.GetPodByLabel(label.Namespace(), label.PodName())
+		if err != nil {
+			t.Errorf("Failed to get Pod %s: %v", label.String(), err)
+		}
+		return realizedPod.Namespace + "/" + realizedPod.Name
+	}
+
+	for _, eval := range npEvaluation.Truth {
+		cmds := append(antctlArgs, "-S", getRealizedPod(eval.Source), "-D", getRealizedPod(eval.Destination))
+		log.Tracef("Running command: %s", strings.Join(cmds, " "))
+		stdout, stderr, err := data.RunCommandFromPod(antreaNamespace, antreaController.Name, "antrea-controller", cmds)
+		assert.NoError(t, err, "Failure -- Command was not successful:\n%s", antctlOutput(stdout, stderr))
+		assert.Containsf(t, stdout, eval.NPName, "Failure -- wrong results for evaluation: NetworkPolicy: %s, expected: %s", stdout, eval.NPName)
+		assert.Contains(t, stdout, eval.Action, "Failure -- wrong results for evaluation: Action: %s, expected: %s", stdout, eval.Action)
```

Comment:
we should do something better here, doesn't the antctl command support structured output with `-o yaml` or `-o json`?

Reply:
@antoninbas An issue arose for `Isolate` again, as currently `Isolate` action is filled in antctl when processing [Table output](https://github.com/antrea-io/antrea/blob/b7dcfb6c151bcf2ba990bdb6bdc23c7ab2c5e5d5/pkg/antctl/transform/networkpolicy/response.go#L188). For Yaml and Json output, they directly marshel response from API, where action is still `nil`.

The only ideal way is to fill `Isolate` in API, but we had this [discussion](https://github.com/antrea-io/antrea/pull/6112#discussion_r1540658586) of not creating a real action for isolation. So I'm thinking just adding isolate `string` to `RuleAction`? This way it's not a real action but still fits the response.

reason: 1) antctl processes all Yaml and Json response together, so I assume it's not ideal to change that
2) I can also modify here in `antreapolicy_test` when comparing, but it would be confusing and Yaml and Json issue  remains

Reply:
So I currently implemented this solution, let me know of any other suggestions.

Reply:
We cannot modify the API as part of an "e2e tests" PR. There needs to be a dedicated PR.

What should have been done when designing the API is that the `Action` field in the `RuleRef` struct used for the NetworkPolicyEvaluation response should not have been of type `crdv1beta1.RuleAction`: https://github.com/antrea-io/antrea/blob/ad7209da108c393b706385f4413c93971e5c03a7/pkg/apis/controlplane/v1beta2/types.go#L480

It should have been either of type `string` or a different type which would have included the `Isolate` action.

The API should also have been versioned with v1alpha1, instead of being made beta right away.

I feel like we should not modify the API at this stage, without introducing a new version (e.g. v1beta3). Given that this is quite a process, we should probably not do that simply for the sake of e2e testing. So we can revert back to your initial version and do string matching, or maybe do some more structured parsing with a regex.

That being said, I feel like other consumers of this API (besides antctl) may find it inconvenient that the "Isolate" action is not part of the API itself. We can wait until we get some feedback before making any change to the API.

Reply:
Understood, thanks for the detailed instruction. I've reverted the change of API, and modified the test case with better processing of Table output.